include/jsonv/serialization_boost.hpp

```diff
@@ -0,0 +1,54 @@
+/** \file jsonv/serialization_boost.hpp
+*  Specialization types for serialization of boost classes and containers.
+*  These are usually not needed unless you are writing your own
+*  \c extractor or \c serializer.
+*
+*  Copyright (c) 2015 by Travis Gockel. All rights reserved.
+*
+*  This program is free software: you can redistribute it and/or modify it under the terms of the Apache License
+*  as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
+*  version.
+*
+*  \author Vladimir Venediktov (vvenedict@gmail.com)
+**/
+
+#ifndef __JSONV_SERIALIZATION_BOOST_HPP_INCLUDED__
+#define __JSONV_SERIALIZATION_BOOST_HPP_INCLUDED__
+
+#include <jsonv/serialization_util.hpp>
+#include <boost/optional.hpp>
```

Comment:
Hi Travis , sounds good will take a look at your suggestion and fix it later tonight.
Also , while I have your attention here , the project I am working with is rather interesting to test your implementation of json.
Real Time Bidding done 90% with json as payload but as usual ad exchanges rarely follow the openRTB standard format.
I will be trying to use your DSL and trying to map all to only one set of structures for all exchanges .
There maybe some challenges there .
I don't have a team yet , just doing everything myself after my day job .
If you have any interest let me know as we can work together .
For reference there is an older implementation of RTB by datacratic 
Here https://github.com/rtbkit
But it brings tons of dependencies plus datacratic non standard structures , so I decided to give it a try .

Talk to you later,
Thanks ,
Vlad.


Sent from my iPhone

> On Jun 4, 2016, at 6:42 AM, Travis Gockel <notifications@github.com> wrote:
> 
> In include/jsonv/serialization_boost.hpp:
> 
> > +*  \c extractor or \c serializer.
> > +*
> > +*  Copyright (c) 2015 by Travis Gockel. All rights reserved.
> > +*
> > +*  This program is free software: you can redistribute it and/or modify it under the terms of the Apache License
> > +*  as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
> > +*  version.
> > +*
> > +*  \author Vladimir Venediktov (vvenedict@gmail.com)
> > +**/
> > +
> > +#ifndef __JSONV_SERIALIZATION_BOOST_HPP_INCLUDED__
> > +#define __JSONV_SERIALIZATION_BOOST_HPP_INCLUDED__
> > +
> > +#include <jsonv/serialization_util.hpp>
> > +#include <boost/optional.hpp>
> Instead of using boost::optional<T> directly, I would prefer doing something like I did with string_view -- allow choosing the optional implementation. There's a small block of macro in string_view.hpp:
> 
> #ifndef JSONV_STRING_VIEW_TYPE
> #   if defined(JSONV_STRING_VIEW_USE_STD) && JSONV_STRING_VIEW_USE_STD
> #       define JSONV_STRING_VIEW_TYPE    std::string_view
> #       define JSONV_STRING_VIEW_INCLUDE <string_view>
> #   elif defined(JSONV_STRING_VIEW_USE_BOOST) && JSONV_STRING_VIEW_USE_BOOST
> #       define JSONV_STRING_VIEW_TYPE    boost::string_ref
> #       define JSONV_STRING_VIEW_INCLUDE <boost/utility/string_ref.hpp>
> #   else
> #       define JSONV_STRING_VIEW_TYPE    jsonv::detail::string_view
> #       define JSONV_STRING_VIEW_INCLUDE <jsonv/detail/string_view.hpp>
> #   endif
> #endif
> Do the same for optional<T> (although don't bother writing a jsonv::detail::optional<T>).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>